### PR TITLE
Fix 1376 juniper junos show interfaces

### LINKED ISSUE #1376: juniper junos_show_interfaces not parsing correctly
##### ISSUE TYPE
 - Template Issue with error and raw data

##### TEMPLATE USING
./ntc_templates/templates/juniper_junos_show_interfaces.textfsm
```
Value Required INTERFACE (\S+)
Value LINK_STATUS (\w+)
Value ADMIN_STATE (\S+)
Value HARDWARE_TYPE (\S+)
Value DESCRIPTION (\w+.*)
Value DESTINATION (\S+)
Value LOCAL (\S+)
Value MTU (\d+|Unlimited)

Start
  ^\s*\S+\s+interface -> Continue.Record
  ^Physical\s+interface:\s+${INTERFACE},\s+${ADMIN_STATE},\s+Physical\s+link\s+is\s+${LINK_STATUS}
  ^.*Description:\s+${DESCRIPTION}
  ^.*ype:\s+${HARDWARE_TYPE},.*MTU:\s+${MTU}.*
  ^\s+Logical\s+interface\s+${INTERFACE}
  ^.*Destination:\s+${DESTINATION},\s+Local:\s+${LOCAL},.*
  ^\s*$$
  ^{master:\d+}

```

##### SAMPLE COMMAND OUTPUT

```
Physical interface: xe-0/1/0, Enabled, Physical link is Up
  Interface index: 153, SNMP ifIndex: 564
  Description: ### My description - remote host - remote location - feature codes - info here ###
  Link-level type: Ethernet, MTU: 9114, MRU: 9122, LAN-PHY mode, Speed: 10Gbps, BPDU Error: None, Loop Detect PDU Error: None, MAC-REWRITE Error: None, Loopback: None, Source filtering: Disabled, Flow control: Enabled, Speed Configuration: 1G
  Pad to minimum frame size: Disabled
  Device flags   : Present Running
  Interface flags: SNMP-Traps Internal: 0x4000
  Link flags     : None
  CoS queues     : 8 supported, 8 maximum usable queues
  Schedulers     : 0
  Current address: 0a:aa:aa:aa:aa:aa, Hardware address: 0a:aa:aa:aa:aa:aa
  Last flapped   : 2022-07-28 21:30:59 GMT (43w0d 15:54 ago)
  Input rate     : 648 bps (1 pps)
  Output rate    : 968 bps (0 pps)
  Active alarms  : None
  Active defects : None
  PCS statistics                      Seconds
    Bit errors                             1
    Errored blocks                         0
  Interface transmit statistics: Disabled
```

##### EXPECTED RESULTS
parsing of description, IP, mac address, etc.

##### ACTUAL RESULTS

```
{'admin_state': 'Enabled',
  'description': '',
  'destination': '',
  'hardware_type': 'Ethernet',
  'interface': 'xe-0/1/0',
  'link_status': 'Up',
  'local': '',
  'mtu': '9114'},
```

### PR DESCRIPTION
**Pull request by diepes**
Fix #1376 , Fixs_1376, Fix_1376

tasks.py add alias test to tests

juniper junos_show_interfaces

- tests/juniper_junos/show_interfaces/juniper_junos_show_interfaces.yml
Sort current test yaml and verify it did not change.

- tests/juniper_junos/show_interfaces/juniper_junos_show_interfaces2.raw
add new test file

- ntc_templates/templates/juniper_junos_show_interfaces.textfsm
Update template.
match all values in 2 tests and end on "-> Error"

**⁇** It seems a bit brittle as Juniper has so many settings that seem to be uniq to each interface type.
Suggest we comment the last line ? ("  ^. -> Error")
Probably need more test examples.

**⁇** the records are generated on blank lines, couldn't see a better way ?

### CODE REVIEW

File: tasks.py [hunk 1]
```diff
@@ -162,7 +162,7 @@ def cli(context):
     context.run(f"{dev}", pty=True)
 
 
-@task(help={"local": "Run locally or within the Docker container"})
+@task(help={"local": "Run locally or within the Docker container"}, aliases=["test"])
```
Comment (jvanderaa) [inline tasks.py:165]: @jmcgill298 an eye on this.

File: tests/juniper_junos/show_interfaces/juniper_junos_show_interfaces2.raw [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  Last flapped   : 2022-07-28 21:30:59 GMT (43w0d 15:54 ago)
+  Input rate     : 648 bps (1 pps)
+  Output rate    : 968 bps (0 pps)
+  Active alarms  : None
+  Active defects : None
+  PCS statistics                      Seconds
+    Bit errors                             1
+    Errored blocks                         0
+  Interface transmit statistics: Disabled
+
```
Comment (jmcgill298) [inline tests/juniper_junos/show_interfaces/juniper_junos_show_interfaces2.raw:21]:
```suggestion
```

File: ntc_templates/templates/juniper_junos_show_interfaces.textfsm [hunk 3]
```diff
@@ -2,17 +2,49 @@ Value Required INTERFACE (\S+)
...
+  ^\s+Input packets
+  ^\s+Output packets
+  ^\s+Security
+  ^\s+Protocol
+  ^\s{6}Flags
+  ^\s{6}Address
+  ^\s+Destination:\s+${DESTINATION},\s+Local:\s+${LOCAL},.*
+  #
   ^{master:\d+}
+  ^. -> Error
```
Comment (jmcgill298) [inline ntc_templates/templates/juniper_junos_show_interfaces.textfsm:50]:
```suggestion
  ^. -> Error

```

File: ntc_templates/templates/juniper_junos_show_interfaces.textfsm [hunk 4]
```diff
@@ -2,19 +2,51 @@ Value Required INTERFACE (\S+)
...
-  ^.*Destination:\s+${DESTINATION},\s+Local:\s+${LOCAL},.*
-  ^\s*$$
+  ^\s{2,4}Flags
+  ^\s+Input packets
+  ^\s+Output packets
+  ^\s+Security
+  ^\s+Protocol
+  ^\s{6}Flags
+  ^\s{6}Address
+  ^\s+Destination:\s+${DESTINATION},\s+Local:\s+${LOCAL},.*
```
Comment (jmcgill298) [inline ntc_templates/templates/juniper_junos_show_interfaces.textfsm:49]:
```suggestion
  ^\s+Destination:\s+${DESTINATION},\s+Local:\s+${LOCAL},.*
  ^\s+MAC:
```

File: ntc_templates/templates/juniper_junos_show_interfaces.textfsm [hunk 5]
```diff
@@ -2,19 +2,52 @@ Value Required INTERFACE (\S+)
...
-Value DESCRIPTION (\w+.*)
+Value DESCRIPTION (.*)
 Value DESTINATION (\S+)
 Value LOCAL (\S+)
 Value MTU (\d+|Unlimited)
 
 Start
-  ^\s*\S+\s+interface -> Continue.Record
+  ## ^\s*\S+\s+[Ii]nterface -> Continue.Record
+  ^\s*$$ -> Record
```
Comment (jmcgill298) [inline ntc_templates/templates/juniper_junos_show_interfaces.textfsm:12]:
```suggestion
  ^\s*\S+\s+[Ii]nterface -> Continue.Record
```

File: ntc_templates/templates/juniper_junos_show_interfaces.textfsm [hunk 6]
```diff
@@ -2,19 +2,52 @@ Value Required INTERFACE (\S+)
...
   ^\s+Protocol\s+inet.*,\s+MTU:\s+${MTU}
-  ^.*Destination:\s+${DESTINATION},\s+Local:\s+${LOCAL},.*
-  ^\s*$$
+  ^\s{2,4}Flags
+  ^\s+Input packets
+  ^\s+Output packets
+  ^\s+Security
+  ^\s+Protocol
+  ^\s{6}Flags
+  ^\s{6}Address
```
Comment (jmcgill298) [inline ntc_templates/templates/juniper_junos_show_interfaces.textfsm:48]:
```suggestion
  ^\s+Flags
  ^\s+Address
  ^\s+Bandwidth
  ^\s+Routing
  ^\s+Max
```

File: ntc_templates/templates/juniper_junos_show_interfaces.textfsm [hunk 7]
```diff
@@ -2,19 +2,52 @@ Value Required INTERFACE (\S+)
...
+  ^\s+Bit errors
+  ^\s+Errored blocks
+  ^\s+Interface transmit statistics
+  #
   ^\s+Logical\s+interface\s+${INTERFACE}
   ^\s+Flags:\s+${LINK_STATUS}\s+\S+\s+\w+\s+(\S+\s+)*Encapsulation:
   ^\s+Protocol\s+inet.*,\s+MTU:\s+${MTU}
-  ^.*Destination:\s+${DESTINATION},\s+Local:\s+${LOCAL},.*
-  ^\s*$$
+  ^\s{2,4}Flags
```
Comment (jmcgill298) [inline ntc_templates/templates/juniper_junos_show_interfaces.textfsm:42]:
```suggestion
  ^\s+Flags
```

**Review by jmcgill298 — APPROVED** [review]

**Pull request merged**